wallet/wallet.h

```diff
@@ -1026,6 +1026,15 @@ void wallet_payment_delete(struct wallet *wallet,
 			   const struct sha256 *payment_hash,
 			   u64 partid);
 
+/**
+ * wallet_payment_delete_by_hash - Remove a payment
+ *
+ * Removes the payment from the database by hash; if it is a MPP payment
+ * it remove all parts with a single query.
+ */
+void wallet_payment_delete_by_hash(struct wallet *wallet,
+				   const struct sha256 *payment_hash);
```

Comment:
Nit: indentation of the followup line should match the `struct` keyword above (might be a GH display issue though).